modules/distributor/forwarder.go

```diff
@@ -189,156 +232,24 @@ func (f *forwarder) watchOverrides() {
 	}
 }
 
-func (f *forwarder) start(_ context.Context) error {
+func (f *generatorForwarder) start(_ context.Context) error {
 	go f.watchOverrides()
 
 	return nil
 
 }
 
-func (f *forwarder) stop(_ error) error {
+func (f *generatorForwarder) stop(_ error) error {
 	close(f.shutdown)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
```

Comment:
does the services.Manager already have a timeout? I'm honestly not super familiar with that code. if it does i'd rather just rely on it to cancel context.